### PR TITLE
Chore: Upgrade golangci-linter

### PR DESCRIPTION
**Pull request by zserge**
This PR upgrades golangci-linter to a newer v1.55.0 release, which got several other linters upgraded.

It will be followed by a few other PRs that reduce the number of linters (replacing them with `revive`) and revert a few "exclude" rules.

### CODE REVIEW

File: .golangci.toml [hunk 1]
```diff
@@ -71,40 +67,29 @@ disable-all = true
 enable = [
   "bodyclose",
   "depguard",
-  "dogsled",
```
Comment (zserge) [inline .golangci.toml:74]: Finally... 🛷

File: .golangci.toml [hunk 2]
```diff
@@ -71,40 +67,29 @@ disable-all = true
...
-  "errcheck",
-  # "gochecknoinits",
-  "goconst",
```
Comment (zserge) [inline .golangci.toml:77]: `goconst` got upgraded and now complains about dozens of strings here and there (usually logs or parts of the DB queries). It makes no sense to implement all its suggesions.

File: .golangci.toml [hunk 3]
```diff
@@ -71,40 +67,29 @@ disable-all = true
...
-  "errorlint",
   "revive",
 ]
 
-# Disabled linters (might want them later)
-# "unparam"
-# "rowserrcheck" # The linter doesn't detect that both Scan and Close also returns the error message returned by Err.
-
-[issues]
-exclude-use-default = false
```
Comment (zserge) [inline .golangci.toml:107]: We basically re-implemented it with exclude-rules below

**Review by mildwonkey — COMMENTED** [review]
🎉

**Review by mildwonkey — APPROVED** [review]

**Review by papagian — APPROVED** [review]

**Pull request closed without merging**